docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,264 @@
+
+> [!NOTE]
+> More details are described in the [`subscribe`](./messages/client/subscribe.md) message.
+
+#### Rollback Handling
+
+In a [CAP](https://en.wikipedia.org/wiki/CAP_theorem) system like Cardano, which balances global consistency with availability, there is a challenge in dealing with rollback events that impact transaction finality. Regardless of whether one uses their own full node or a server provider for transaction submission - if such events are not properly handled with the help of monitoring the rolling window of [k blocks](https://plutus-apps.readthedocs.io/en/latest/plutus/explanations/rollback.html) and respective resubmission, there is no guarantee that the transaction is finalized, eg. becomes part of the immutable pefix of the Cardano blockchain.
+
+Since our wallet-optimized protocol defines that server-sent messages include the `point` message partial, the client can check if its last `point` is greater than any subsequent message `point` from the server, without the need to introduce a `rollback` specific message partial. This assumes that the client keeps a local copy of the volatile segment of the blockchain defined by the security parameter.
+
+#### Connection Management
+
+Clients are expected to reconnect on error, when providing invalid `point`s for which a server cannot find an intersection.
+Clients may send multiple [`subscribe`](./messages/client/subscribe.md) messages for different accounts, but subsequently take responsibility to coordinate server-sent events by time to their possibly differently synchronized accounts. When receiving a message that references a `point` that the first account has already seen, the client only applies it for its second account for example.
+
+### Versioning
+
+We propose to version message partials individually. In our protocol, the client sends its list of supported versions as part of the initial [`subscribe`](./messages/client/subscribe.md) message. Upon receiving this information, the server selects the highest mutually supported version and associates it with the client.
+This method requires the server to implement version-based message construction, allowing it to tailor its responses to each client's supported version.
+
+### Encoding
+
+Our proposed communication protocol **adopts blockchain-native encoding standards** to ensure compatibility and efficiency. This approach leverages existing conventions familiar to clients while minimizing conversion overhead on the server-side. (see [CPS - Fragile API](../../CPS/CPS-XXXX/README.md#fragile-api))
+
+#### Address Encoding
+
+Address encoding varies depending on the blockchain network. In Cardano the protocol servers bech32 encoded addresses.
+
+### Extensions
```

Comment:
We need to define how extensions will be supported under this standard. Best to take inspiration from https://github.com/cardano-foundation/CIPs/pull/462.